test/validate_test.go

```diff
@@ -0,0 +1,277 @@
+package test
+
+import (
+	"context"
+	"fmt"
+	"testing"
+
+	"github.com/GoogleCloudPlatform/terraform-validator/converters/google"
+	"github.com/GoogleCloudPlatform/terraform-validator/tfgcv"
+	"github.com/GoogleCloudPlatform/terraform-validator/tfplan"
+	"github.com/hashicorp/terraform/terraform"
+	"google.golang.org/api/cloudresourcemanager/v1"
+	"google.golang.org/api/option"
+)
+
+func TestValidate(t *testing.T) {
+	cfg := configure(t)
```

Comment:
All of this setup code should be moved after the testcase definition, it's hinders readability for it to be separated by all the testcase defs.

Reply:
It turns out it can't be moved. The function `configure(t)` is required to be run before the test case constructions, as it read the `TEST_PROJECT`, `TEST_CREDENTIAL` and `TEST_POLICY` from environment and the project name (`TEST_PROJECT`) is required to build the test case.